src/meteors/visualize/shap/plots.py

```diff
@@ -0,0 +1,691 @@
+
+from meteors.shap import HyperSHAP, SHAPExplanation
+
+import shap
+import numpy as np
+import pandas as pd
+
+import matplotlib.pyplot as plt
+
+
+def validate_observation_index(
+    observation_index: int | None, explanation: SHAPExplanation, require_local_explanation: bool = False, plot_type=None
+) -> int | None:
+    """function validates the observation index for the explanation.
+    It checks if the observation index is an integer or None and within the bounds of the explanation.
+    In case the single observation is required (local explanation), the function checks if the selected observation can be determined (the input explanation is also local).
+
+    In case the single observation is not required (global explanation), the function allows for None as the observation index, which means that the explanation is for all observations.
+
+    Args:
+        observation_index (int | None): a selected observation index. Defaults to None.
+        explanation (SHAPExplanation): An explanation object, from which the observations will be selected.
+        require_local_explanation (bool, optional): A flag that determines whether the plot function accepts global explanations. Defaults to False.
+        plot_type (str, optional): The type of the plot. Defaults to None. Used to print the customized ValueError message.
+
+    Raises:
+        TypeError: In case the observation index is not an integer or None.
+        ValueError: Whether the integer observation index is out of bounds.
+        ValueError: if the observation index is not specified, but explanation contains multiple observations and the plot supports only local explanations.
+
+    Returns:
+        int | None: parsed observation index.
+    """
+    if not isinstance(observation_index, int) and observation_index is not None:
+        raise TypeError(f"Expected int or None as observation index, but got {type(observation_index)}")
+    if isinstance(observation_index, int):
+        if explanation.is_local_explanation and observation_index != 0:
+            raise ValueError(
+                "Passed explanation is for a single observation only - index must be 0 or None for single observation explanations"
+            )
+        if not explanation.is_local_explanation and (
+            observation_index < 0 or observation_index >= explanation.data.shape[0]
+        ):
+            raise ValueError(
+                f"Observation index out of bounds: {observation_index}. The explanation contains {explanation.explanations.values.shape[0]} observations"
+            )
+
+    if require_local_explanation:
+        if observation_index is None and not explanation.is_local_explanation:
+            raise ValueError(
+                f"The plot of type {plot_type} only supports local explanations. \nPassed explanation contains multiple observations and no observation index specified."
+            )
+        if observation_index is None:
+            observation_index = 0
```

Comment:
Should it be outside of the if?

Reply:
What exactly are you asking for? In case the local explanation is not required by the visualization method, we don't need to parse the `observation_index`.